src/base/Atomics.hh

```diff
@@ -0,0 +1,35 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file Atomics.hh
+//---------------------------------------------------------------------------//
+#ifndef base_Atomics_hh
+#define base_Atomics_hh
+
+#include "Assert.hh"
+#include "Macros.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Add to a value, returning the original value.
+ */
+template<class T>
+CELER_FORCEINLINE_FUNCTION T atomic_add(T* address, T value)
```

Comment:
Are you assuming that we will always be using CUDA > Version 6?  Otherwise, we will have to specialize for double because they didn't have hardware atomics for double below that version.  I think that's fine by the way as all our machines have CUDA v10 or greater now.  We may want to check the version at configure time if we aren't though to enforce this.

Reply:
I think that's a safe assumption -- 6.0 was released february 2014. Or is this a device capability? I'll look into having the minimum requirement.

Reply:
According to the cuda manual, 
> The 32-bit floating-point version of atomicAdd() is only supported by devices of compute capability 2.x and higher.
so hopefully we get compile-time error from cuda if the hardware is really ancient?

I can't find a straightforward way of guaranteeing the cuda toolkit is > 6,  but  I don't think any of the hardware we're targeting uses such an old version, so I'd say let's skip trying to verify those versions until someone actually hits an error from an incompatible old version.